### PR TITLE
JENKINS-56703 full testcoverage including exceptions

### PR DESCRIPTION
**Pull request by LorMun**

### CODE REVIEW

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.AssertionsForClassTypes.*;
+
+/**
+ * Testclass for LinesDataSet.
+ *
+ * @author Lorenz Munsch
+ */
+class LinesDataSetTest {
+
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: No empty lines

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        LinesDataSet linesDataSet  = new LinesDataSet();
+        assertThat(linesDataSet.getXAxisSize()).isEqualTo(0);
+
+        Map<String, Integer>  dataSetSeries = new HashMap<>();
+        linesDataSet.add("Testlabel", dataSetSeries);
+        assertThat(linesDataSet.getXAxisSize()).isEqualTo(1);
+    }
+
+    @Test
+    void getXAxisLabels() {
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: Combine with previous test case

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        LinesDataSet linesDataSet  = new LinesDataSet();
+        Map<String, Integer>  dataSetSeries = new HashMap<>();
+        linesDataSet.add("Testlabel", dataSetSeries);
+        assertThat(linesDataSet.getXAxisLabels()).isEqualTo(Collections.singletonList("Testlabel"));
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: Use collections assertions (see https://joel-costigliola.github.io/assertj/assertj-core-quick-start.html).

```
assertThat(linesDataSet.getXAxisLabels()).containsOnly("foo")
```

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    void getDataSetIds() {
+        LinesDataSet linesDataSet  = new LinesDataSet();
+        Map<String, Integer>  dataSetSeries = new HashMap<>();
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: extract to method (and optionally use `Maps.fixedSize.of`)

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        dataSetSeries.put("Eins", 1);
+        dataSetSeries.put("Zwei", 2);
+        dataSetSeries.put("Drei", 3);
+        dataSetSeries.put("Vier", 4);
+        linesDataSet.add("Testlabel", dataSetSeries);
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: extract `TestLabel` to constant

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        assertThat(linesDataSet.getDataSetIds().size()).isEqualTo(4);
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: use native collection assertions, see comment above
```
assertThat(collection).hasSize(...).contains(...)
```

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+/**
+ * Testclass for LinesDataSet.
+ *
+ * @author Lorenz Munsch
+ */
+class LinesDataSetTest {
+
+
+    @Test
+    void getXAxisSize() {
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: Typically the name starts with should and explains the rationale behind test

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        dataSetSeries.put("Zwei", 2);
+        dataSetSeries.put("Drei", 3);
+        dataSetSeries.put("Vier", 4);
+        linesDataSet.add("Testlabel", dataSetSeries);
+        assertThat(linesDataSet.getDataSetIds().size()).isEqualTo(4);
+        assertThat(linesDataSet.getDataSetIds().contains("Eins")).isEqualTo(true);
+        assertThat(linesDataSet.getDataSetIds().contains("Zwei")).isEqualTo(true);
+        assertThat(linesDataSet.getDataSetIds().contains("Drei")).isEqualTo(true);
+        assertThat(linesDataSet.getDataSetIds().contains("Vier")).isEqualTo(true);
+        assertThat(linesDataSet.getDataSetIds().contains("Fuenf")).isEqualTo(false);
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: The last statement does not make sense, just use `containsOnly`

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        dataSetSeries.put("Eins", 1);
+        dataSetSeries.put("Zwei", 2);
+        dataSetSeries.put("Drei", 3);
+        dataSetSeries.put("Vier", 4);
+        linesDataSet.add("Testlabel", dataSetSeries);
+        assertThat(linesDataSet.hasSeries("Eins")).isEqualTo(true);
+    }
+
+    @Test
+    void getSeries() {
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: Combine with previous test

File: src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        dataSetSeries.put("Vier", 4);
+        linesDataSet.add("Testlabel", dataSetSeries);
+        assertThat(linesDataSet.getSeries("Eins")).isEqualTo(Collections.singletonList(1));
+    }
+
+    @Test
+    void getSeriesWithMissingEntry() {
+        LinesDataSet linesDataSet  = new LinesDataSet();
+        assertThat(linesDataSet.hasSeries("Eins")).isEqualTo(false);
+        Map<String, Integer>  dataSetSeries = new HashMap<>();
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/LinesDataSetTest.java]: Not required, just use empty map

**Comment by LorMun** [comment]
I tried to resolve all discussions!

**Pull request merged**